backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -1122,11 +1126,24 @@ private class InteropTransformer(val context: Context, override val irFile: IrFi
 
         return argument.statements.last() as? IrFunctionReference
     }
+
+    val IrValueParameter.isDispatchReceiver: Boolean
+        get() = parent.let {
+            when {
```

Comment:
```kotlin
get() = when (val parent = this.parent) {
```

Reply:
It seems to be impossible to use such a captured variable in _conditions_ of the `when` operator (only in actions):
```
InteropLowering.kt: (1132, 13): Incompatible types: Boolean and IrDeclarationParent
```

Reply:
```kotlin
val IrValueParameter.isDispatchReceiver: Boolean
    get() = when (val parent = this.parent) {
       is IrClass -> true
       is IrFunction -> parent.dispatchReceiverParameter == this
       else -> false
    }
```
?